group_project_v2/project_api/api_implementation.py

```diff
@@ -301,17 +304,19 @@ def get_workgroup_by_id(self, group_id):
         return WorkgroupDetails(**response)
 
     @memoize_with_expiration()
-    def get_user_workgroup_for_course(self, user_id, course_id):
+    def get_user_workgroup_for_course(self, user_id, course_id, fallback=None):
```

Comment:
@jbzdak I don't think the fallback should be here. The idea is to have this class only query the API and return the actual result, converted to typed DTOs. So, if no group was found, it should return None and let caller decide what to do about it.